RecoLocalTracker/SiPixelRecHits/plugins/PixelCPEFastESProducer.cc

```diff
@@ -89,8 +89,8 @@ void PixelCPEFastESProducer::fillDescriptions(edm::ConfigurationDescriptions& de
   // specific to PixelCPEFastESProducer
   desc.add<std::string>("ComponentName", "PixelCPEFast");
   desc.add<edm::ESInputTag>("MagneticFieldRecord", edm::ESInputTag());
-  desc.add<bool>("useLAAlignmentOffsets", false);
-  desc.add<bool>("DoLorentz", false);
+  desc.add<bool>("useLAAlignmentOffsets", false)->setComment("deprecated");
+  desc.add<bool>("DoLorentz", false)->setComment("deprecated");
```

Comment:
shouldn't this be `addOptional`?
I was looking at https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp#Optional_Parameters
by the context I would think that addObsolete would be a more proper naming, but it seems like it's used for other purposes.
The context here is related to https://github.com/cms-sw/cmssw/pull/33792#discussion_r635894241 `the parameters in question be labelled "obsolete" in fillDescriptions of the plugins, such that configs containing them would still work`

@makortel @Dr15Jones 
please check/advise.
Thank you.

Reply:
`addOptional()` without a default value is a closest feature for declaring obsolete parameters in the `fillDescriptions()`, i.e. such parameter will not be part of the generated cfi file, configuration validation does nothing such parameter is missing, and defining such parameter does not cause error.

Reply:
@slava77 addressed at https://github.com/cms-sw/cmssw/pull/33792/commits/64ab2c307e4be193ada9aa6eb5b51252fd2893da